api/src/plugins/code-flow-auth.ts

```diff
@@ -80,6 +84,34 @@ const codeFlowAuth: FastifyPluginCallback = (fastify, _options, done) => {
     }
   );
 
+  fastify.decorate(
+    'addUserIfAuthorized',
+    async function (req: FastifyRequest, reply: FastifyReply) {
+      const tokenCookie = req.cookies.jwt_access_token;
+      if (tokenCookie) {
+        const unsignedToken = req.unsignCookie(tokenCookie);
+        if (unsignedToken.valid) {
+          const jwtAccessToken = unsignedToken.value;
+          try {
+            jwt.verify(jwtAccessToken!, JWT_SECRET);
+            const {
+              accessToken: { created, ttl, userId }
+            } = jwt.decode(jwtAccessToken!) as { accessToken: AccessToken };
+            const valid = isBefore(Date.now(), Date.parse(created) + ttl);
+            if (valid) {
+              const user = await fastify.prisma.user.findUnique({
+                where: { id: userId }
+              });
+              if (user) req.user = user;
+            }
+          } catch {
+            return send401(reply, TOKEN_INVALID);
```

Comment:
Not sure if I need to return an error or not here since we are just checking if user is authorized and pass the user object.